stage #2/module #2. Web, HTTP(s) methods, Java EE Servlets/lesson #1 Web & HTTP/content.md

```diff
@@ -1,175 +1,169 @@
 HTTP (Hypertext Transfer Protocol)
-- HTTP is a protocol that clients and servers on the web to communicate. 
-- It is similar to other internet protocols such as SMTP (Simple Mail Transfer Protocol) and FTP (File Transfer Protocol.
-- HTTP is a stateless protocol i.e it supports only one request per connection. This means that with HTTP the clients connect to the server to send one request and then disconnect. This mechanism allows more users to connect to a given server over a period of time.
-- The client sends an HTTP request and the server answers with an HTML page to the client, using HTTP.
+- HTTP is a protocol that clients and servers on the web use to communicate. 
+- It is similar to other internet protocols such as SMTP (Simple Mail Transfer Protocol) and FTP (File Transfer Protocol).
+- HTTP is a stateless protocol i.e. it supports only one request per connection (although since HTTP/1.1 "keep-alive" connections allow you to send several HTTP requests over one TCP connection). This means that with HTTP the clients connect to the server to send one request and then disconnect. This mechanism allows more users to connect to a given server over a period of time.
+- The client sends an HTTP request and the server answers with an HTTP response containing an HTML page, JSON data or something else requested.
 
 In an extremely simplified form, the HTTP architecture can be represented as follows:
 
 ![http architecture](media/http_architecture.png)
 
 HTTP messages are composed of textual information encoded in ASCII, and span over multiple lines.
-Web developers rarely craft these textual HTTP messages themselves: software, a Web browser, proxy, or Web server, perform this action. 
+Web developers rarely craft these textual HTTP messages themselves: software, web browsers, proxy, or web servers perform this action. 
 
-## HTTP Message structure
-HTTP messages are how data is exchanged between a server and a client. There are two types of messages: request and response.
-
-HTTP requests and responses share similar structure and are composed of:
-1. A <b>start-line</b> describing the requests to be implemented, or its status of whether successful or a failure. This start-line is always a single line.
-2. An optional set of HTTP <b>headers</b> specifying the request, or describing the body included in the message.
-3. An <b>empty line</b> indicating the end of meta-information.
-4. An optional <b>body</b> containing data associated with the request (like content of an HTML form), or the document associated with a response. The presence of the body and its size is specified by the start-line and HTTP headers.
-
-The start-line and HTTP headers of the HTTP message are collectively known as the <b>head of the request</b>, whereas its body is known as the <b>payload</b>.
-
-![http message structure](media/http_message_structure.png)
-
-### HTTP Request structure
+## HTTP Request structure
 
 The start-line of HTTP request contains three elements:
```

Comment:
Please highlight the main parts: start-line, HTTP headers, body to make them more memorable